misago/users/avatars/gallery.py

```diff
@@ -66,7 +68,12 @@ def load_avatar_galleries():
 
 
 def set_avatar(user, image):
-    store.store_new_avatar(user, Image.open(image.path))
+    try:
+        response = requests.get(image.url)
```

Comment:
`requests` should be unnecessary here - value of `File` and `Image` model fields is `File` instance that has `open()` method that delegates opening/reading operation to your configured `STATICFILES_STORAGE`:

```
with image.open() as img:
    store.store_new_avatar(user, Image.open(img))
```

Could you please say if that code works? I don't have S3 to test it out myself.